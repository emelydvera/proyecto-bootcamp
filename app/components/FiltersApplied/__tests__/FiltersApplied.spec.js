const React = require('react');
const { render, screen, fireEvent } = require('@testing-library/react');
const FiltersApplied = require('..');
const { data } = require('../../../../mocks/test/get/https/api.mercadolibre.com/sites/MCO/search?limit=10&offset=0&page=1&q=tablet.json')

const UrlGenerator = require('../../../utils/urlGenerator');
jest.mock('../../../utils/urlGenerator')

describe('FiltersApplied', () => {


  beforeEach(() => {
    UrlGenerator.mockClear();
  })

  const filters = [
    {
      "id": "shipping",
      "name": "Tipo de entrega",
      "type": "text",
      "values": [
        {
          "id": "fulfillment",
          "name": "Full"
        }
      ]
    }
  ]

  it('Should render correctly', () => {
    const urlGenerator = new UrlGenerator();
    const { asFragment } = render(<FiltersApplied filters={filters} urlGenerator={urlGenerator} />);
    expect(asFragment()).toMatchSnapshot();
  })

  it('Should render the tag of the filter applied', () => {
    const urlGenerator = new UrlGenerator();
    render(<FiltersApplied filters={filters} urlGenerator={urlGenerator} />);
    const tagFilter = screen.queryByText(/Full/);
    expect(tagFilter).toBeInTheDocument()
  })

  it('Should call the urlGenerator method to remove the filter when the tag delete button is clicked', () => {
    const urlGenerator = new UrlGenerator();
    render(<FiltersApplied filters={filters} urlGenerator={urlGenerator} />);
    const deleteButton = screen.queryByRole(/button/);
    expect(urlGenerator.removeFilter).toBeCalledTimes(0);
    fireEvent.click(deleteButton)
    expect(urlGenerator.removeFilter).toBeCalledTimes(1);
    expect(urlGenerator.removeFilter).toBeCalledWith("shipping");
  })

  it('Should set the filter category when the category is deleted ', () => {
    const urlGenerator = new UrlGenerator();
    render(<FiltersApplied filters={[data.filters[0]]} urlGenerator={urlGenerator} />);
    const deleteButton = screen.queryByRole(/button/);
    expect(urlGenerator.setFilter).toBeCalledTimes(0);
    fireEvent.click(deleteButton)
    expect(urlGenerator.setFilter).toBeCalledTimes(1);
    expect(urlGenerator.setFilter).toBeCalledWith("category", "MCO432947");
  })

  it('Should not call the urlGenerator method when the category path_from_root length is less than 2  ', () => {
    const urlGenerator = new UrlGenerator();

    const filterCategory = [
      {
        "id": "category",
        "name": "Categorías",
        "type": "text",
        "values": [
          {
            "id": "MCO82067",
            "name": "Tablets",
            "path_from_root": [
              {
                "id": "MCO1648",
                "name": "Computación"
              },
            ]
          }
        ]
      }
    ]

    render(<FiltersApplied filters={filterCategory} urlGenerator={urlGenerator} />);
    const deleteButton = screen.queryByRole(/button/);
    expect(urlGenerator.setFilter).toBeCalledTimes(0);
    fireEvent.click(deleteButton)
    expect(urlGenerator.setFilter).toBeCalledTimes(0);
  })

})