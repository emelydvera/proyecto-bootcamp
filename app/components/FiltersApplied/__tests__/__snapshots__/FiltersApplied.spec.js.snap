// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersApplied Should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="andes-tag andes-tag--small andes-tag__color--default"
    >
      <div
        class="andes-tag__label"
      >
        Full
      </div>
      <button
        aria-label="Eliminar filtro: Tipo de entrega: Full"
        class="andes-tag__close"
        type="button"
      >
        <span
          aria-hidden="true"
          class="andes-tag__icon-container"
        >
          <svg
            class="andes-tag__close-icon andes-tag__svg-fill"
            height="12"
            viewBox="0 0 16 16"
            width="12"
          >
            <path
              d="M8.594.552l.855.842L5.87 5.022 9.45 8.6l-.849.848-3.572-3.572-3.521 3.572-.855-.842L4.18 5.028.552 1.4l.849-.848 3.621 3.62L8.594.553z"
              fill-opacity="0.45"
              transform="translate(3 3)"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
